File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i = len(nums)//2
        for i in range(len(nums)):
            if nums[i] == target:
                return i
            elif nums[i] > target:
                i -= i//2
            else:
                i += i//2

        return -1

        

File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = nums[0];
        for (int i = 0; i < nums.size(); i++)
        {
            int currentSum = 0;
            for (int j = i; j < nums.size(); j++)
            {
                currentSum += nums[j];

                if (currentSum > sum)
                {
                    sum = currentSum;
                }

            }
            
        }

        return sum;

    }
};
